pkg/networking/util.go

```diff
@@ -0,0 +1,76 @@
+package networking
+
+import (
+	"net"
+	"strings"
+
+	"encoding/json"
+	"fmt"
+	"os/exec"
+
+	"github.com/vishvananda/netlink"
+	"k8s.io/api/core/v1"
+)
+
+type Link struct {
+	Type string           `json:"type"`
+	IP   string           `json:"ip"`
+	Name string           `json:"name"`
+	MAC  net.HardwareAddr `json:"mac"`
+}
+
+func GetInterfaceFromIP(ip string) (iface netlink.Link, err error) {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+	for _, iface := range interfaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return nil, err
+		}
+		for _, addr := range addrs {
+			if ip == strings.Split(addr.String(), "/")[0] {
+				return netlink.LinkByName(iface.Name)
+			}
+		}
+
+	}
+	return nil, nil
+}
+
+func GetNodeInternalIP(node *v1.Node) (ip string) {
+	for _, addr := range node.Status.Addresses {
+		if addr.Type == v1.NodeInternalIP {
+			return addr.Address
+		}
+	}
+	return ""
+}
+
+type IntrospectorInterface interface {
+	GetLinkByIP(ip string, pid string) (*Link, error)
+}
+
+type introspector struct {
+	toolDir string
+}
+
+func NewIntrospector(toolDir string) IntrospectorInterface {
```

Comment:
maybe fast fail with a fatal error if we detect network helper binary isn't present?